In [117]:
%run runaway_functionsv3
cl = Cluster("Berkeley_97")

# cl.stars_in_region()
# cl.stars_in_region()

In [118]:
cl.fast_stars_in_region()

<Table length=258>
   RA_ICRS_1       DE_ICRS_1    ...         rmDE               v_pec       
      deg             deg       ...       mas / yr             km / s      
    float64         float64     ...       float64             float64      
--------------- --------------- ... -------------------- ------------------
339.78276898862  59.07086926297 ...   1.1141785192489624 20.387853576815186
340.11072199405  58.92710174793 ...  -2.2078214807510372 45.388586118778576
339.74109632951  58.83600736330 ... -0.29382148075103753  18.97672144868768
339.43148475774  58.95033027843 ...  -0.1278214807510376 29.836092292894257
339.78212101782  58.86686000942 ...   -2.180821480751038 33.273753811137496
339.41403033977  58.99750133848 ...   -2.629821480751038  37.86769328936914
339.56494819769  58.99344914887 ...  -1.8448214807510377 33.734465330421564
339.91038508587  58.88525715423 ...   -3.185821480751038  33.26576709756339
339.99450594962  59.24767717932 ...  -2.3898214807510376  40.38407666376765
            ...             ... ...                  ...                ...
340.15726824235  59.20539197373 ...  -1.4808214807510378 21.548353039404883
339.91118684081  58.89372945236 ...   2.6931785192489626  39.44035656488606
339.73360156668  58.85404017708 ...   0.5691785192489625  34.21715074718525
339.70907225506  58.98738634674 ...   0.2811785192489624 33.072632233064006
340.26572770448  59.08593627063 ...  -0.8998214807510374  24.62298560253607
339.58283285271  59.12168876192 ...   1.0421785192489623  34.76925835274061
340.21515251556  58.84935165812 ...  -2.5838214807510376 52.001672329183634
340.05473039811  58.91413229797 ... -0.09082148075103769 31.736175728742044
339.68148663862  59.07323997618 ...  -1.7678214807510377 28.014568088692844
339.68163569354  58.76229337605 ...   3.2251785192489626  43.03961268320435

In [119]:
g = cl.gießler_input()
g



<Table length=259>
TypeInput      RA           DE      ...    pmDE    e_pmDE         Source      
                                    ...  mas / yr mas / yr                    
  int32      str12        str12     ...  float64  float32         str21       
--------- ------------ ------------ ... --------- -------- -------------------
        1 22 39 23.976 59 00 16.560 ...    -1.884    0.099         Berkeley_97
        1 22 39 07.865 59 04 15.129 ...    -0.770    0.034 2008431406040225664
        1 22 40 26.573 58 55 37.566 ...    -4.092    0.027 2008379900791042048
        1 22 38 57.863 58 50 09.627 ...    -2.178    0.012 2008378212857607680
        1 22 37 43.556 58 57 01.189 ...    -2.012    0.046 2008409862483646080
        1 22 39 07.709 58 52 00.696 ...    -4.065    0.013 2008381309540033024
        1 22 37 39.367 58 59 51.005 ...    -4.514    0.016 2008410201776259968
        1 22 38 15.588 58 59 36.417 ...    -3.729    0.013 2008407491661705216
        1 22 39 38.492 58 53 06.926 ...    -5.070    0.011 2008378625176070272
      ...          ...          ... ...       ...      ...                 ...
        1 22 40 37.744 59 12 19.411 ...    -3.365    0.056 2008439205700851840
        1 22 39 38.685 58 53 37.426 ...     0.809    0.052 2008378835638877056
        1 22 38 56.064 58 51 14.545 ...    -1.315    0.050 2008381172101075456
        1 22 38 50.177 58 59 14.591 ...    -1.603    0.048 2008429962930927488
        1 22 41 03.775 59 05 09.371 ...    -2.784    0.055 2008390517950147968
        1 22 38 19.880 59 07 18.080 ...    -0.842    0.219 2008435426124887552
        1 22 40 51.637 58 50 57.666 ...    -4.468    0.060 2008366736706369920
        1 22 40 13.135 58 54 50.876 ...    -1.975    0.055 2008379827767031680
        1 22 38 43.557 59 04 23.664 ...    -3.652    0.056 2008434459751900928
        1 22 38 43.593 58 45 44.256 ...     1.341    0.052 2008354337144968576

#    RA                  DEC               Parallax        RV Maxw.        PMRA              PMDEC      ID           Remarks
# ---------------------------------------------------------------------------------------------------------------------------
1  00 14 17.75       +47 46 33.4        0.5631 0.0563     0 0 1        19.3    1.8       -19.7    1.5   B0011+47     (aus NS_no_GCs.txt, Zeile 3)
1  00 34 08.8703     -07 21 53.409        0.93   0.08     0 0 1       10.37   0.08      -11.13   0.16   B0031-07     (aus NS_no_GCs.txt, Zeile 4)
1  00 40 32.3899     +57 16 24.833        0.11   0.04     0 0 1       12.39   0.05        -5.4    0.1   B0037+56     (aus NS_no_GCs.txt, Zeile 5)
1  00 55 45.3981     +51 17 24.601        0.35   0.06     0 0 1       10.47   0.07       -17.4    0.1   B0052+51     (aus NS_no_GCs.txt, Zeile 6)
1  01 02 32.9914     +65 37 13.416         0.4   0.04     0 0 1        9.24   0.07        1.77   0.15   B0059+65     (aus NS_no_GCs.txt, Zeile 7)
1  01 08 22.5049     +66 08 34.499        0.47   0.03     0 0 1      -32.75   0.03       35.15   0.04   B0105+65     (aus NS_no_GCs.txt, Zeile 8)
1  01 08 08.347016   -14 31 50.1871        4.2    1.4     0 0 1          75      3      -152.5    1.7   J0108-1431   (aus NS_no_GCs.txt, Zeile 9)
1  01 25 01.074328   -23 27 08.1349        1.4    0.3     0 0 1       37.14   0.08        10.5    0.2   J0125-2327   (aus NS_no_GCs.txt, Zeile 10)
1  01 34 18.6824     -29 37 17.042        0.04  0.004     0 0 1          17      1          -9      2   J0134-2937   (aus NS_no_GCs.txt, Zeile 11)
1  01 47 44.6434     +59 22 03.284        0.47   0.07     0 0 1       -6.39   0.09         3.8   0.08   B0144+59     (aus NS_no_GCs.txt, Zeile 12)
1  01 51 22.7179     -06 35 02.987        0.23   0.09     0 0 1        10.7    0.1       -5.38   0.07   B0148-06     (aus NS_no_GCs.txt, Zeile 13)

In [131]:
myrun = Table.read('./Clusters/Berkeley_97/Berkeley_97_runaways_all.tsv', format='ascii.ecsv')
myrun

<Table length=6>
Temp. Est    RA_ICRS_1       DE_ICRS_1    ...   e_RPmag     e_BP-RP  
    K           deg             deg       ...                        
 float64      float64         float64     ...   float32     float32  
--------- --------------- --------------- ... ----------- -----------
  14950.0 339.89581711423  58.99547917199 ...    0.004229    0.008081
   9894.0 339.82327510265  59.02791551662 ...    0.004356    0.009985
   8341.0 339.76625014965  58.97239278067 ...    0.004317    0.008444
   7654.0 339.72574294861  58.90086167091 ...    0.004717     0.01149
   7654.0 340.04434455421  59.06446992114 ...    0.005277     0.01166
   7413.0 339.86590073968  59.01069960562 ...    0.004242     0.01225

In [135]:
gsources = [2008378693895483648,
            2008431406040225664,
            2008383817801137152,
            2008432058875274496,
            2008384573715370624,
            2008381034662369920,
            2008406185991730560,
            2008384504995669248,
            2008391033346185856,
            2008387421269978112,
            2008385192190655616,
            2008383126302250624,
            2008391411303287808,
            2008436319478858368,
            2008430169089361536,
            2008385020393773056,
            2008432814789510400,
            2008384814233513472,
            2008405979833246592,
            2008430306528315136,
            2008389933834808192,
            2008384779873564160,
            2008406289070887424,
            2008382374691897728,
            2008378693895483648
            ]

gtable = Table({'Source':gsources})
fs = cl.fast_stars_in_region()
fs = join(fs,gtable, keys='Source', join_type='inner')
fs.add_column(np.zeros_like(fs['BP-RP']), name='Temp. Est', index=0)
fs.write('test.tsv', format='ascii.ecsv', overwrite=True)

In [138]:
def make_gießler_input(table):
    g = Table()
    g['TypeInput'] = np.ones_like(table['e_Plx'].value).astype(int)
    g['RA'] = table['SkyCoord'].ra.to_string(unit='hourangle', sep=' ', precision=3, pad=True)
    g['DE'] = table['SkyCoord'].dec.to_string(unit='degree', sep=' ', precision=3, pad=True)
    g['Plx'] = table['rgeo'].to(u.mas, u.parallax())
    g['e_Plx'] = table['e_Plx']
    g['RV'] = np.zeros_like(table['e_Plx'].value).astype(int)
    g['e_RV'] = np.zeros_like(table['e_Plx'].value).astype(int)
    g['RVdist'] = np.zeros_like(table['e_Plx'].value).astype(int)
    g['pmRA'] = table['pmRA']
    g['e_pmRA'] = table['e_pmRA']
    g['pmDE'] = table['pmDE']
    g['e_pmDE'] = table['e_pmDE']
    g['Source'] = table['Source'].astype(str)
    return g
make_gießler_input(fs).write(f'temp.tsv', format='csv', delimiter='\t', overwrite=True)
